lib/omise/OmiseSearch.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+require_once dirname(__FILE__).'/res/OmiseApiResource.php';
+
+/**
+ * This class is not intended to be used directly from client code.
+ *
+ * @see OmiseCharge::search()
+ */
+class OmiseSearch extends OmiseApiResource
+{
+    const ENDPOINT = 'search';
+
+    private $dirty = true;
+    private $scope = null;
+    private $query = '';
+    private $conditions = array();
+    private $page = 1;
+    private $order = 'chronological';
+
+    public static function scope($scope)
+    {
+        return new OmiseSearch($scope);
+    }
+
+    protected function __construct($scope, $publickey = null, $secretkey = null)
+    {
+        parent::__construct($publickey, $secretkey);
+        $this->dirty = true;
+        $this->scope = $scope;
+    }
+
+    public function query($query)
+    {
+        $this->dirty = true;
+        $this->query = $query;
+        return $this;
+    }
```

Comment:
@iwat I really like this implementation, also wanna enhance these methods a bit, how about something like,
```php 
public function page($page)
{
    return $this->mergeAttribute('page', $page);
}

protected function mergeAttribute($key, $value)
{
    $this->dirty = true;
    $this->$key  = $value;

    return $this;
}
```
_note, I tried run test and it passed._

Or, how about this approach 👇 (just idea, I haven't made test to prove yet).
```php
private $scope = null;
private $query = '';
private $conditions = array();
private $page = 1;
private $order = 'chronological';

// to

private $attributes;

protected function __construct($scope, $publickey = null, $secretkey = null)
{
    parent::__construct($publickey, $secretkey);
    return $this->mergeAttribute('scope', $scope);
}

protected function mergeAttribute($key, $value)
{
    $this->dirty            = true;
    $this->attributes[$key] = $value;

    return $this;
}

// Now we can use $this->attributes; with various situation.
// i.e. $querystring = http_build_query($this->attributes);
```